students/ShujinCao/lesson03/assignment/test_basic_operations.py

```diff
@@ -0,0 +1,86 @@
+"""
+tests
+"""
+import pytest
+from basic_operations import *
+from customer_model import Customer
+
+good_customer2 = {
+    'customer_id': '2WDIll8IDO',
+    'name': 'Suzie2',
+    'last_name': 'Edgar',
+    'home_address': '122 Brown St., Seattle, WA 98121',
+    'phone_number': '122-333-444',
+    'email_address': '2suzie@gmail.com',
+    'status': True,
+    'credit_limit': 40.00}
+
+good_customer = {
+    'customer_id': 'WDIll8IDO',
+    'name': 'Suzie',
+    'last_name': 'Edgar',
+    'home_address': '122 Brown St., Seattle, WA 98121',
+    'phone_number': '123-333-444',
+    'email_address': 'suzie@gmail.com',
+    'status': True,
+    'credit_limit': 40.00}
+
+bad_customer = {'customer_id':'8675309',
+                'name': 'Jenny',
+                'last_name': 'Igotyonumber',
+                'home_address': 'Beverly Hills Plaza, Compton, Ca. 90210',
+                'phone_number': '198-765-4321',
+                'email_address': 'Jenny_Igotyonumber@gmail.com',
+                'status': True,
+                'credit_limit': 40.00}
+
+def test_add_customer():
+    # add_customer(**good_customer)
+    test_customer = Customer.get(Customer.customer_id==good_customer['customer_id'])
+    assert test_customer.email_address == good_customer['email_address']
```

Comment:
I'm thinking that you commented out the add_customer, because the customer was still there from the previous run. You should make fixtures for before/after your tests that create your table before the run and delete it after. https://docs.pytest.org/en/latest/fixture.html You can look in my student directory for an example of doing it with unittest.